File: RegexTester/ShortestComparer.cs
namespace RegexTester
{
    using System;
    using System.Collections.Generic;
    using System.Text.RegularExpressions;

    public class ShortestComparer : IComparer<Match>
    {
        public int Compare(Match x, Match y)
        {
            return x.Length.CompareTo(y.Length);
        }
    }
}

File: RegexTester/ShortestComparer.vb
Imports System
Imports System.Collections.Generic
Imports System.Text.RegularExpressions

Namespace RegexTester
    Public Class ShortestComparer
        Implements IComparer(Of Match)
        ' Methods
        Public Function [Compare](ByVal x As Match, ByVal y As Match) As Integer Implements IComparer(Of Match).Compare
            Return x.Length.CompareTo(y.Length)
        End Function

    End Class
End Namespace

